### PR TITLE
Docs: Documentation for the AG Grid

### PR DESCRIPTION
**Pull request by maxschulz-COL**
## Description
This PR contains the documentation for the AG Grid feature

### Open ToDos
- [ ] Exchange Screenshots
- [ ] Align with latest version of implementation (if changed)

@AnnMarieW I took your stylings on the CSS PR as an example on how to style and individual AG Grid - the "advanced" docs examples. Thanks!

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-core/docs/pages/user_guides/custom_tables.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
-# How to create custom tables
+# How to create custom Tables
```
Comment (huong-li-nguyen) [inline vizro-core/docs/pages/user_guides/custom_tables.md:1]: Would stay consistent with the rest of the docs e.g. "How to create custom charts"
```suggestion
# How to create custom tables
```

File: vizro-core/docs/pages/user_guides/custom_tables.md [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
-If you want to use the [`Table`][vizro.models.Table] model to and to create a custom [table](table.md) you can create your own custom table, e.g. when requiring computations that can be controlled by parameters.
+If you want to use the [`Table`][vizro.models.Table] model to create a custom [table](table.md), e.g. when requiring computations that can be controlled by parameters, then the below approach is recommended.
```
Comment (huong-li-nguyen) [inline vizro-core/docs/pages/user_guides/custom_tables.md:3]: Would remove the example - I think it's too specific and I didn't understand immediately what kind of table that is 😄 

```suggestion
If you want to use the [`Table`][vizro.models.Table] model to create a custom [table](table.md) then the below approach is recommended.
```

File: vizro-core/docs/pages/user_guides/table.md [hunk 3]
```diff
@@ -17,6 +17,195 @@ To add a [`Table`][vizro.models.Table] to your page, do the following:
 
 See below for an overview of currently supported table functions.
 
+### Dash AG Grid
+
+The [Dash AG Grid](https://dash.plotly.com/dash-ag-grid) is an interactive table/grid component designed for viewing, editing, and exploring large datasets.
+It is Vizro's recommended table implementation.
```
Comment (huong-li-nguyen) [inline vizro-core/docs/pages/user_guides/table.md:23]:
```suggestion
The [Dash AG Grid](https://dash.plotly.com/dash-ag-grid) is an interactive table/grid component designed for viewing, editing, and exploring tabular data. It is Vizro's recommended table implementation.
```

File: vizro-core/docs/pages/user_guides/table.md [hunk 4]
```diff
@@ -17,6 +17,195 @@ To add a [`Table`][vizro.models.Table] to your page, do the following:
...
+        import vizro.plotly.express as px
+        from vizro import Vizro
+        from vizro.tables import dash_ag_grid
+
+        df = px.data.gapminder().query("year == 2007")
+
+        page = vm.Page(
+            title="Example of a Dash AG Grid",
+            components=[
+                vm.Table(id="table", title="Dash AG Grid", figure=dash_ag_grid(data_frame=df)),
```
Comment (huong-li-nguyen) [inline vizro-core/docs/pages/user_guides/table.md:48]: I think we've aligned that we won't add an ID to the examples/docs anymore unless required

```suggestion
                vm.Table(title="Dash AG Grid", figure=dash_ag_grid(data_frame=df)),
```

File: vizro-core/docs/pages/user_guides/table.md [hunk 5]
```diff
@@ -17,6 +17,195 @@ To add a [`Table`][vizro.models.Table] to your page, do the following:
...
+          title: Example of a Dash AG Grid
+        ```
+    === "Result"
+        [![Table]][Table]
+
+    [Table]: ../../assets/user_guides/table/table.png
+
+#### Styling/Modifying the Dash AG Grid
+
+As mentioned above, all [parameters of the Dash AG Grid](https://dash.plotly.com/dash-ag-grid/reference) can be entered as keyword arguments. Below you can find
```
Comment (huong-li-nguyen) [inline vizro-core/docs/pages/user_guides/table.md:80]: As soon as this PR is merged: https://github.com/mckinsey/vizro/pull/267

Let's replace the demo example with the ag-grid one and add the ag-grid to the features-dashboard 👍

File: vizro-core/src/vizro/tables/dash_aggrid.py [hunk 6]
```diff
@@ -1,15 +1,21 @@
...
-    )
+def dash_ag_grid(data_frame: pd.DataFrame = None, **kwargs):
+    """Standard `dash-ag-grid.AgGrid`."""
+    defaults = {
+        "columnDefs": [{"field": col, "filter": True} for col in data_frame.columns],
+        "className": "ag-theme-alpine vizro",
+        "rowData": data_frame.to_dict("records"),
+        "dashGridOptions": {"rowHeight": 40},
+        "defaultColDef": {"resizable": True, "sortable": True},
+        "columnSize": "sizeToFit",
```
Comment (huong-li-nguyen) [inline vizro-core/src/vizro/tables/dash_aggrid.py:18]: I think some of these can be moved to CSS e.g. rowHeight

What does columSize do?

**Review by huong-li-nguyen — COMMENTED** [review]
Thanks for adding this in! Have a few suggestions but looks great already! 🚀 

Will tests be added separately?

**Comment by maxschulz-COL** [comment]
Closed with https://github.com/mckinsey/vizro/pull/324 now open - implemented most comments there

**Pull request closed without merging**